packages/orama/src/components/defaults.ts

```diff
@@ -23,22 +23,30 @@ export async function getDocumentIndexId(doc: Document): Promise<string> {
   return await uniqueId()
 }
 
-export async function validateSchema<S extends Schema = Schema>(doc: Document, schema: S): Promise<boolean> {
+export async function validateSchema<S extends Schema = Schema>(doc: Document, schema: S): Promise<string | undefined> {
   for (const [prop, type] of Object.entries(schema)) {
+    const value = doc[prop]
```

Comment:
You'd be using `typeof value` at least twice here, so maybe it's worth it to instanciate a new variable `const valueType = typeof value`?

Reply:
it is used below in the for code

Reply:
done